File: sources/fsharp/06_char_count/main.fs
// task 06 char_count — expected output: 10000000
// build: dotnet build -c Release    run: dotnet run

// The whole 100 MB text is built in one operation, not by appending.
let text = String.replicate 10000000 "abcdefghij"

let mutable count = 0L
let len = text.Length
let mutable i = 0
while i < len do
    let ch = text.[i]
    if ch = 'a' then
        ()
    elif ch = 'e' then
        ()
    elif ch = 'h' then
        count <- count + 1L
    else
        ()
    i <- i + 1

printfn "%d" count

File: sources/dart/06_char_count/main.dart
// task 06 char_count — expected output: 10000000
// build: dart compile exe -o prog main.dart (aot; jit has no build step)    run: ./prog (aot) | dart main.dart (jit)

void main() {
  // Built once, by repeating the whole block in one operation.
  final String text = 'abcdefghij' * 10000000;

  int count = 0;
  for (int i = 0; i < text.length; i++) {
    final int ch = text.codeUnitAt(i);
    if (ch == 0x61) {
      // 'a': skip
    } else if (ch == 0x65) {
      // 'e': skip
    } else if (ch == 0x68) {
      count += 1; // 'h'
    }
    // anything else: skip
  }
  print(count);
}
